src/Tribe/Event_Scheduler.php

```diff
@@ -0,0 +1,203 @@
+	 * @param mixed $permanently_delete
+	 *
+	 * @since TBD
+	 */
+	public function __construct( $move_to_trash = null, $permanently_delete = null ) {
+		$this->trash_new_date = $move_to_trash;
+		$this->del_new_date   = $permanently_delete;
+	}
+
+	/**
+	 * Retrieves the new user-defined value for trashPastEvents option
+	 * and defines it as the trash_new_date variable.
+	 *
+	 * @param $mixed - the value for the trashPastEvents option
+	 *
+	 * @since TBD
+	 */
+	public function trash_set_new_date( $trash_new_value ) {
+		$this->trash_new_date = $trash_new_value;
+	}
+
+	/**
+	 * Retrieves the new user-defined value for deletePastEvents option
+	 * and defines it as the del_new_date variable.
+	 *
+	 * @param $mixed - the value for the deletePastEvents option
+	 *
+	 * @since TBD
+	 */
+	public function delete_set_new_date( $del_new_value ) {
+		$this->del_new_date = $del_new_value;
+	}
+
+	/**
+	 * Schedules the hooks to delete and move old events to trash
+	 * These hooks will be executed daily.
+	 *
+	 * @since TBD
+	 */
+	public function add_hooks() {
+		if ( ! wp_next_scheduled( self::TRASH_CRON_HOOK ) && $this->trash_new_date != null ) {
+			wp_schedule_event( time(), 'daily', self::TRASH_CRON_HOOK );
+		}
+
+		if ( ! wp_next_scheduled( self::DEL_CRON_HOOK ) && $this->del_new_date != null ) {
+			wp_schedule_event( time(), 'daily', self::DEL_CRON_HOOK );
+		}
+
+		if ( $this->trash_new_date != null ) {
+			add_action( self::TRASH_CRON_HOOK, array( $this, 'move_old_events_to_trash' ), 10, 0 );
+		}
+
+		if ( $this->del_new_date != null ) {
+			add_action( self::DEL_CRON_HOOK, array( $this, 'permanently_delete_old_events' ), 10, 0 );
+		}
+
+		add_action( 'tribe_events_blog_deactivate', array( $this, 'trash_clear_scheduled_task' ) );
+		add_action( 'tribe_events_blog_deactivate', array( $this, 'delete_clear_scheduled_task' ) );
+	}
+
+	/**
+	 * Removes the hooks
+	 *
+	 * @since TBD
+	 */
+	public function remove_hooks() {
+		remove_action( self::TRASH_CRON_HOOK, array( $this, 'move_old_events_to_trash' ), 10, 0 );
+		remove_action( self::DEL_CRON_HOOK, array( $this, 'permanently_delete_old_events' ), 10, 0 );
+	}
+
+	/**
+	 * Un-schedules all previously-scheduled cron jobs for tribe-trash-event-cron
+	 *
+	 * @since TBD
+	 */
+	public function trash_clear_scheduled_task() {
+		wp_clear_scheduled_hook( self::TRASH_CRON_HOOK );
+	}
+
+	/**
+	 * Un-schedules all previously-scheduled cron jobs for tribe-del-event-cron
+	 *
+	 * @since TBD
+	 */
+	public function delete_clear_scheduled_task() {
+		wp_clear_scheduled_hook( self::DEL_CRON_HOOK );
+	}
+
+	/**
+	 * Selects events to be moved to trash or permanently deleted.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $month - The value chosen by user to purge all events older than x months
+	 *
+	 * @return array $post_ids - an array of event Post_IDs with the Event End Date older than $month
+	 */
+	public function select_events_to_purge( $month ) {
+		/** @var wpdb $wpdb */
+		global $wpdb;
```

Comment:
something brand new we introduced in the last mr is a way to access `$wpdb` with some "safety checks" (to help prevent fatal errors in case a theme or other plugin is messing with the global in an uncouth way)

try changing `global $wpdb` to this:
```
$wpdb = tribe_get_global_query_object();
```
then *test* your code again after making that change; sometimes it causes issues. if it does, then just leave it as `global $wpdb`...but if using the "safer" method doesn't cause any issues, use that 👍

Reply:
Hey @ggwicz, I ran a test and replaced `global $wpdb` with `$wpdb = tribe_get_global_query_object();` as suggested, but the feature doesn't work with this change, so I'm leaving it as `global $wpdb` for now.

Reply:
This is good as is, we are using the tribe_get_global_query_object(); to get $wp_query, not $wpdb.